modules/mongodb/src/main/java/org/testcontainers/containers/MongoDbContainer.java

```diff
@@ -0,0 +1,152 @@
+package org.testcontainers.containers;
+
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.IOException;
+
+/**
+ * Constructs a single node MongoDB replica set for testing transactions.
+ * <p>To construct a multi-node MongoDB cluster, consider the <a href="https://github.com/silaev/mongodb-replica-set/">mongodb-replica-set project on GitHub</a>
+ * <p>Tested on a Mongo DB version 4.0.10+ (that is the default version if not specified).
+ *
+ * @author Konstantin Silaev on 9/30/2019
+ */
+@Slf4j
+public class MongoDbContainer extends GenericContainer<MongoDbContainer> {
+    private static final int CONTAINER_EXIT_CODE_OK = 0;
+    static final int MONGODB_INTERNAL_PORT = 27017;
+    private static final int AWAIT_INIT_REPLICA_SET_ATTEMPTS = 30;
+    private static final String MONGODB_VERSION_DEFAULT = "4.0.10";
+    private static final String LOCALHOST = "localhost";
+    private static final String MONGODB_DATABASE_NAME_DEFAULT = "test";
+
+    public MongoDbContainer() {
+        super("mongo:" + MONGODB_VERSION_DEFAULT);
+    }
+
+    public MongoDbContainer(@NonNull final String dockerImageName) {
+        super(dockerImageName);
+    }
+
+    public String getReplicaSetUrl() {
+        if (!isRunning()) {
+            throw new IllegalStateException("MongoDbContainer should be started first");
+        }
+        return String.format(
+            "mongodb://%s:%d/%s",
+            getContainerIpAddress(),
+            getMappedPort(MONGODB_INTERNAL_PORT),
+            MONGODB_DATABASE_NAME_DEFAULT
+        );
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        initReplicaSet();
+        logReplicaSetStatus();
+    }
+
+    @SneakyThrows(value = {IOException.class, InterruptedException.class})
+    private void logReplicaSetStatus() {
+        log.debug(
+            "REPLICA SET STATUS:\n{}",
+            execInContainer(buildMongoEvalCommand("rs.status()")).getStdout()
+        );
+    }
+
+    @Override
+    protected void configure() {
+        withExposedPorts(MONGODB_INTERNAL_PORT);
+        withCommand("--replSet", "docker-rs");
+        waitingFor(
+            Wait.forLogMessage(".*waiting for connections on port.*", 1)
+        );
+    }
+
+    private String getMongoReplicaSetInitializer() {
+        final String containerIpAddress = getContainerIpAddress();
+        final int containerPort = LOCALHOST.equals(containerIpAddress)
```

Comment:
this looks like a red flag. Could you please explain in more and if you really think it should be like that, please also add comments in the code

Reply:
It stands for MongoDB replica set initialization specific for working elsewhere. Please, take a look at the table below, I've added similar Javadoc to the method.
 
| Difference  | local Docker example | local Docker host running tests from inside a container with mapping the Docker socket example |
| --- | --- | --- | 
| a host string to initialize a replica set  |  localhost:27017 <br>**Despite the fact that Docker allocates 33538 (for instance) as a random port for a container**|  172.17.0.1:33542 |
| a url to use with a Java Mongo driver  | mongodb://localhost:33538/test  |  mongodb://172.17.0.1:33542/test |

Reply:
I've seen the table. Yet, I don't understand *the problem*. In my previous experience, matching "localhost" was wrong most of the times. What problem are you solving here? Links to MongoDB's docs may help too.

Reply:
Unfortunately, it's not good covered in docs. That's why I created this PR to simplify things.
[Here is one related ticket](https://jira.mongodb.org/browse/SERVER-1889) though. If you want, I can write a test proving that the table above is correct.

Just in case, [we discussed localhost concern with](https://github.com/testcontainers/testcontainers-java/pull/1961#discussion_r333715893) @rnorth some time ago

Reply:
If it is not covered, could you please describe the problem and how your workaround works? It is important, because we need to clearly understand what are we doing here and what are the potential implications. Thanks.

Reply:
When it comes to a localhost, an attempt to initialize a replica set with a randomly assigned Docker port leads to a "No host described in new configuration 1 for replica set docker-rs maps to this node" error.

As a workaround, I use MongoDB internal port 27017 in a replica set initialization command.  To be honest, it’s also possible not to set any host or port **only** for localhost for Standalone mode.  
However, my goal is to also support _local Docker host running tests from inside a container with mapping the Docker socket and remote Docker daemon_ which require real port to set in a replica set initialization command.
To sum up, there is a need to know whether an environment a localhost or not. What’s why I use localhost in my code.

There is no problem with resolvable names via DNS or an OS host file [see more details here](https://serverfault.com/a/816961), but my code does not use any of them.

If you are really willing to avoid the “localhost“ word in this PR, I can refactor the code to do the following double attempt:

1. Try to initialize replica set running on localhost (with port 27017, no "localhost" word);
2. If 1. fails, try to initialize replica on _local Docker host running tests from inside a container with mapping the Docker socket or remote Docker daemon_ (with a real randomly assigned Docker port).

Personally speaking, this "double attempt" solution doesn’t seem to be a good one for 2 reasons:

- Less cleaner and readable code comparing to the current solution;
- One redundant attempt (see 1. above) for those that run on _local Docker host running tests from inside a container with mapping the Docker socket or remote Docker daemon_.

Please, let me know about your decision.

Reply:
>  local Docker host running tests from inside a container with mapping the Docker socket or remote Docker daemon

wait, aren't you always initialize RS **inside** MongoDbContainer? In which `localhost` never equals to tests' `localhost`?

Reply:
Always inside a MongoDbContainer, I'm talking here about 2 cases:

- mapping docker.sock and running our tests inside a Docker container [for examle](https://github.com/silaev/mongodb-replica-set/blob/master/docker-compose-it-test.yml)
- using remote Docker daemon, for instance with $env:DOCKER_HOST="tcp://192.168.1.150:2376"

As a result, these 2 cases lead to a real IP when it comes to getContainerIpAddress(), not to localhost (for example 172.17.0.1).

Reply:
@bsideup Refactored the code to use a default replica set configuration. No need for localhost anymore. Also corrected one of my previous answers.

Reply:
done